src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,473 @@ const ActorInit En_Stone_heishi_InitVars = {
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STAND_UP:
+            if (this->endFrame <= currentFrame) {
+                func_801A3098(NA_SE_PL_DUMMY_290);
+                this->bottleDisplay = EN_STONE_BOTTLE_NONE;
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_STAND_UP);
+                this->DrinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STANDING:
+            if (this->endFrame <= currentFrame) {
+                this->textIndex = 5;
+                func_80151938(play, sEnStoneHeishiTextIds[this->textIndex]);
+                player->actor.textId = sEnStoneHeishiTextIds[this->textIndex];
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_STAND_HAND_ON_HIP);
+                this->action = EN_STONE_ACTION_2;
+                this->actionFunc = func_80BC9680;
+            }
+            break;
+    }
+}
+
+void EnStoneheishi_GiveItemPrize(EnStoneheishi* this, PlayState* play) {
+    func_801477B4(play);
+
+    if (gSaveContext.save.inventory.items[gItemSlots[0x45]] == 0x45) {
+        Actor_PickUp(&this->actor, play, GI_RUPEE_BLUE, 300.0f, 300.0f);
+    } else {
+        Actor_PickUp(&this->actor, play, GI_MASK_STONE, 300.0f, 300.0f);
+    }
+
+    this->action = EN_STONE_ACTION_5;
+    this->actionFunc = func_80BC9D28;
+}
+
+void func_80BC9D28(EnStoneheishi* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_HasParent(&this->actor, play)) {
+        this->actor.parent = NULL;
+        this->textIndex++;
+        this->actor.textId = sEnStoneHeishiTextIds[this->textIndex];
+        gSaveContext.save.weekEventReg[0x29] |= 0x40;
+        Actor_ProcessTalkRequest(&this->actor, &play->state);
+        func_800B8500(&this->actor, play, 400.0f, 400.0f, PLAYER_AP_MINUS1);
+        this->actionFunc = func_80BC9E50;
+    } else if (gSaveContext.save.inventory.items[gItemSlots[0x45]] == 0x45) {
+        Actor_PickUp(&this->actor, play, GI_RUPEE_BLUE, 300.0f, 300.0f);
+    } else {
+        Actor_PickUp(&this->actor, play, GI_MASK_STONE, 300.0f, 300.0f);
+    }
+}
+
+void func_80BC9E50(EnStoneheishi* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        func_80151BB4(play, 0x35);
+        func_80151BB4(play, 0x12);
+        this->action = EN_STONE_ACTION_2;
+        this->actionFunc = func_80BC9680;
+    } else {
+        func_800B8500(&this->actor, play, 400.0f, 400.0f, PLAYER_AP_MINUS1);
+    }
+}
+
+void EnStoneheishi_Update(Actor* thisx, PlayState* play) {
+    EnStoneheishi* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actionFunc(this, play);
+
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 20.0f, 50.0f, 29);
```

Comment:
```suggestion
    Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
```
Since this last argument is a flag, better to leave it in hex